### ISSUE TITLE
CICE Config

### ISSUE THREAD

**Issue by anton-seaice** [ACCESS-NRI/access-cm2-configs#2]
in cice_in.nml - `istep0`

`istep0` is set to 17280 (see https://code.metoffice.gov.uk/trac/roses-u/browser/c/y/3/3/9/trunk/app/cice/rose-app.conf?rev=305501#L324)

`istep0` is the number of timesteps of the initialization time. Its not clear if we can set it to 0, or should be udpating `istep0` for every new restart. Its also possible that setting `use_restart_time = .true.` would mean that this value is ignored.
 
this doesn't appear to affect the cice calendar during the model run (this should be checked) but does mean the cice log file prints some strange / incorrect dates during initialization.

**Comment by MartinDix** [comment]
17280 is strange. It's the number of time steps in 360 days, so perhaps a hangover from when we were using the Met Office climate 360 day years?

From a CM2 PIcontrol starting at 950-01-01, `ice_diag_d ` prints
`istep1:     17280    idate:   9501227    sec:         0`

The CICE namelists are very confusing to me. `input_ice.nml` has `runtime` and `runtime0`. `cice_in.nml` has `days_per_year`, `istep0` and `npt`. The CM2 drivers update all these ***except*** `istep0`, https://github.com/ACCESS-NRI/access-cm2-drivers/blob/main/src/fix_cice_namelist.py

It looks like ESM1.5 only updates runtime and doesn't change `npt` or `days_per_year` for leap years, so these must not matter? E.g. for a PI control run where 003 is a leap year
```
% grep 'runtime =' output00?/ice/*nml
output000/ice/input_ice.nml:    runtime = 31536000
output001/ice/input_ice.nml:    runtime = 31536000
output002/ice/input_ice.nml:    runtime = 31536000
output003/ice/input_ice.nml:    runtime = 31622400

% grep 'npt =' output00?/ice/*nml
output000/ice/cice_in.nml:    npt = 8760.0
output001/ice/cice_in.nml:    npt = 8760.0
output002/ice/cice_in.nml:    npt = 8760.0
output003/ice/cice_in.nml:    npt = 8760.0

% grep 'days_per_year =' output00?/ice/*nml
output000/ice/cice_in.nml:    days_per_year = 365
output001/ice/cice_in.nml:    days_per_year = 365
output002/ice/cice_in.nml:    days_per_year = 365
output003/ice/cice_in.nml:    days_per_year = 365
```

ESM `istep0` increments by `npt`. Looks like there's some double counting.

From output000
`istep1:         0    idate:   1010101    sec:         0`
From output001
`istep1:      8760    idate:   1030101    sec:         0`
so it's showing a date a year further on then it should. Leap years later mess it up more.

**Comment by anton-seaice** [comment]
Thanks Martin

For CM2

I think the idea is that `year_init` doesn't get changed (it's the start year of the experiment) and `istep0` is relative to `year_init`. Saying that - we must be overwritting or ignoring `istep0` somewhere after it's read from the namelist because the cice model dates are correct. So we might just be able to set `istep0` in the `cice_in.nml` and the problem goes away.

For ESM - @blimlim has spent a bit of time looking into this recently, I believe the latest payu should work! (It looks like CICE4 might be figuring out the leap year internally e.g. https://github.com/ACCESS-NRI/cice4/blob/e7549ebd2044690a432cc67c1317c81cb194b750/source/ice_calendar.F90#L421C7-L421C12)

**Comment by blimlim** [comment]
Yeah, the calendar settings for CICE coupled in access are pretty confusing. In this case, it looks like the wrong date appearing  is then corrected by the model. From what I can tell, the only effect is the error in the logging:
```
(init_calendar) idate0 =      9500101
(calendar)  idate =      9501227
(calendar)  idate =      9500101
```
I think if we set `istep0 = 0`, it should correct the logs.  I'll add some details of what I think is happening in the next comment.

**Comment by blimlim** [comment]
**Background on the namelist parameters:**
This is based on ESM1.5, but I think the definitions also work in CM2:

#### `input_ice.nml`:
- `init_date`: The experiment "initialisation" date, a fixed basis time that other quantities are relative to.
- `inidate`: The start time for a given simulation.
- `runtime0`: The number of seconds already simulated. Should match the seconds between the `init_date` and `inidate`.
- `runtime`: The number of seconds to be simulated in the new run.

#### `cice_in.nml`:
- `year_init`: Similar to `init_date` in `input_ice.nml`. A basis year that other quantities are relative to.
- `istep0`: The number of timesteps between the beginning of `year_init` and the experiment start (analogous to `runtime0`).
- `npt`: The number of timesteps to simulate in the new run (analogous to `runtime`, though this is mostly ignored in ESM1.5 and possibly CM2).

There's also a time in the restart file that should match `runtime0`.

**CM2 calendar during initialisation**

The `iceout085` log contains:
```
------------------------------------------------------------------------
<SW: first call to calendar>
 (init_calendar) idate0 =      9500101
 (init_calendar) idate0 (-corrected-) =      9500101
 (calendar) ttime =    31104000.0000000     
 (calendar) nyr, year_init, month, mday =            1         950          12
          27
 (calendar)  idate =      9501227

------------------------------------------------------------------------
<SW: second call to calendar>
 (calendar) ttime =    31104000.0000000     
 (calendar) nyr, year_init, month, mday =            1         950          12
          27
 (calendar)  idate =      9501227

------------------------------------------------------------------------
<SW: third call to calendar>
 CICE (cice_init) 2      time =            0   31104000.0000000     
 CICE (cice_init) 2  runtime0 =            0  0.000000000000000E+000
 (calendar) ttime =    1800.00000000000     
 (calendar) nyr, year_init, month, mday =            1         950           1
           1
 (calendar)  idate =      9500101

 CICE (cice_init) 3     time =            0   1800.00000000000     
 CICE (cice_init) 3 runtime0 =            0  0.000000000000000E+000
 CICE (cice_init) 3    idate =            0     9500101
```
which comes from [this section of `cice_init`](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L153-L220).

There's three calls to `calendar`. The first two print out the wrong date while the third is correct. Given an input `ttime` in seconds, [`calendar(ttime)` will print out](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/source/ice_calendar.F90#L295-L296) the date corresponding to 
```
inidate + ttime
```
where all the proper arithmetic is done to add `ttime` seconds to`inidate`, which is the simulation start date set in `input_ice.nml`. 

The [first call](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L153
) uses input `time`:
```
call calendar(time)
```
which looks like it's [initialised as `istep0+dt`](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/source/ice_calendar.F90#L133), i.e. `17280`. This gives us the extra 360 days when the date is printed to the logs.

The [second call is from `init_restart`](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L359), which also uses `time` as the parameter and prints out the wrong date.

After this , `time` is [reset here](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L192) and adjusted [here](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L208):
```fortran
     time = runtime0   
     ...
     time = time + dt   
```
Which gives us `time = runtime0+dt`. The above lines come under a couple of if statements, but for CM2, we have `jobnum = 2` and `runtype = 'continue'` being set by the driver, which should cause the lines to run. 


Finally the [third call to `calendar` uses argument `time - runtime0`](https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_InitMod.F90#L214). Hence the argument would be `dt`. This then prints out the correct date, given by
```
inidate + dt
```


During the actual run, `time` is incremented by `dt` each timestep, and `calendar` is called with argument `time - runtime0` (see https://github.com/ACCESS-NRI/cice5/blob/0a84ab156217cc8e782affbe7f13071e59a230d8/drivers/access/CICE_RunMod.F90#L171-L174)
```fortran
          time = time + dt       ! determine the time and date
          time_sec = time_sec + dt
          call calendar(time-runtime0) 
```
For the rest of the run, the model should then be using the correct date.
